tracer/test/Datadog.Trace.Tests/Util/RequestDataHelperTests.cs

```diff
@@ -125,6 +129,70 @@ public void GivenADangerousBody_WhenGetValues_HelperAvoidsException()
         }
     }
 
+    [Fact]
+    public void BuildUrl_WhenUrlHasNotBeenAccessed_ShouldReturnUrlAndResetUrlField()
+    {
+        var request = CreateHttpRequest("GET");
+        var urlField = typeof(HttpRequest).GetField("_url", BindingFlags.NonPublic | BindingFlags.Instance);
+
+        // Ensure _url is null before the test
+        // this will mean that Url will need to be built
+        urlField.GetValue(request).Should().BeNull();
+
+        var url = RequestDataHelper.BuildUrl(request);
+
+        url.Should().NotBeNull();
+        url.ToString().Should().Be("http://127.0.0.1/test/test.aspx");
+
+        // Verify that _url was reset to null
+        var urlValueAfter = urlField.GetValue(request) as Uri;
+        urlValueAfter.Should().BeNull();
+    }
+
+    [Fact]
+    public void BuildUrl_WhenUrlIsResetAndHttpRequestIsModified_ShouldReturnUpdatedUrl()
```

Comment:
Nice tests 👍 I wonder if it's worth having a test demonstrating the defective behaviour too, for documentation if nothing else 🤔 Absolutely not wedded it to that though, just wondering

Reply:
Added a test to demonstrate it 👍